packages/entity-server/src/routes/hierarchy/middleware/middleware.ts

```diff
@@ -34,16 +22,33 @@ export const attachContext = async (req: HierarchyRequest, res: Response, next:
 
     const hierarchy = await req.models.entityHierarchy.findOne({ name: hierarchyName });
     const entity = await req.models.entity.findOne({ code: entityCode });
+    if (!hierarchy || !entity) {
+      throwNoAccessError(hierarchyName, entityCode);
+    }
+
+    const rootEntity = entity.isProject()
+      ? entity
+      : await req.models.entity.findOne({ code: hierarchy.name });
+
+    const allowedCountries = (await rootEntity.getChildren(hierarchy.id))
+      .filter(e => req.accessPolicy.allows(e.country_code || undefined))
+      .map(e => e.country_code)
+      .filter(notNull);
```

Comment:
Might be cool to de-dupe here as well, in case the children of the project aren't countries themselves, and we end up with like 100 `FJ` codes. Not necessary as I think the assumption that project children are countries is baked in elsewhere (and also there's no broken logic as a result of having duplicates in the allowed countries list)

Reply:
Ooh good pickup!